lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -102,22 +102,24 @@ def server_version
       kubectl.server_version
     end
 
-    def initialize(namespace:, context:, current_sha:, template_dir:, logger:, kubectl_instance: nil, bindings: {},
+    def initialize(namespace:, context:, current_sha:, template_dirs:, logger:, kubectl_instance: nil, bindings: {},
```

Comment:
I was trying to think of ways to avoid this breaking API change, but the options that occur to me aren't pretty:
* Keep `template_dir:,` but let it accept an array and do `@template_dirs = Array(template_dir)` in the initializer
* have both `template_dir:,` and `template_dirs:`, either the initializer merges them or it raises an error if they're both specified

It kinda irks me to make a breaking API change like this soon before we make another one (with Krane, which is a single clean cutover in a way). I'd be kinda ok with something like the second option above as a result. If we keep it breaking though, let's make it very prominent in the changelog.